upt_macports/upt_macports.py

```diff
@@ -176,6 +180,47 @@ def create_package(self, upt_pkg, output=None):
             pkg_cls = pkg_classes[upt_pkg.frontend]
         except KeyError:
             raise upt.UnhandledFrontendError(self.name, upt_pkg.frontend)
-
         packager = pkg_cls()
         packager.create_package(upt_pkg, output)
+
+    def pkg_exist(self):
+        prefix = {
+            'pypi': 'py-',
+            'rubygems': 'rb-',
+            'cpan': 'p5-',
+        }
+        port_name = prefix.get(self.upt_pkg.frontend, '') + self.upt_pkg.name
+        self.logger.info(f'Checking MacPorts tree for port {port_name}')
+        cmd = f'port info --version {port_name}'
+        port = subprocess.getoutput(cmd)
+        self.logger.info(f'Current Upstream version is {self.upt_pkg.version}')
+        if port[:5] == 'Error':
+            self.logger.info(f'{port_name} not found in MacPorts tree')
+            self.curr_ver = ''
```

Comment:
We are never going to re-use this, right?

And even in the "else:" part of the code, it feels like you are adding info to attributes only so that it can be used in the caller. This makes the code hard to read, as it becomes "spaghetti code", I think.

Maybe pkg_exist could return a tuple (such as (boolean:exist, str:version))? Or just the version: a string when the package exists, None otherwise? Of course the name of the method should be updated.